### PR TITLE
Ftr layout

### PR DESCRIPTION
**Pull request by VB2webD**
#1

### CODE REVIEW

File: __tests__/index.test.jsx [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 import React from "react";
 import { render, screen } from "@testing-library/react";
-import Home from "../pages/index";
+import Home from "../pages/Index";
```
Comment (djfarly) [inline __tests__/index.test.jsx:9]:
```suggestion
import Home from "../pages/index";
```
was correct before ;)

File: components/layout/Footer.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        <a>Favourites</a>
+      </ActiveLink>
+      <ActiveLink href="/checkout" activeClassName="active">
+        <a>Checkout</a>
+      </ActiveLink>
+    </StyledFooter>
+  );
+}
+
+const StyledFooter = styled.ul`
```
Comment (djfarly) [inline components/layout/Footer.jsx:24]: Why is this an `ul`? Maybe `footer` might be more appropriate

File: pages/checkout.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import Layout from "../components/layout/Layout";
+
+const Checkout = () => {
+  return (
+    <Layout title={"Checkout"} hasFooter={true}>
```
Comment (djfarly) [inline pages/checkout.js:6]: You can use the shorthand for boolean react props here:
```suggestion
    <Layout title="Checkout" hasFooter>
```

If your layout doesn't need a footer you can just get rid of the property for that case which for react means that the value for that prop is `undefined` which is [falsy](https://developer.mozilla.org/de/docs/Glossary/Falsy) enough for most cases. 

> also note the `=""` syntax for string props

File: pages/favourites.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import Layout from "../components/layout/Layout";
+
+const Favourites = () => {
+  return (
+    <Layout title={"Favourites"} hasFooter={true}>
```
Comment (djfarly) [inline pages/favourites.js:6]:
```suggestion
    <Layout title="Favourites" hasFooter>
```
see above

File: pages/index.js [hunk 5]
```diff
@@ -1,3 +1,10 @@
+import React from "react";
+import Layout from "../components/layout/Layout";
+
 export default function Home() {
-  return <h1>Hello World</h1>;
+  return (
+    <Layout title={"Index"} hasFooter={true}>
```
Comment (djfarly) [inline pages/index.js:6]:
```suggestion
    <Layout title="Index" hasFooter>
```

File: pages/shop.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import Layout from "../components/layout/Layout";
+
+const Shop = () => {
+  return (
+    <Layout title={"Shop"} hasFooter={true}>
```
Comment (djfarly) [inline pages/shop.js:6]:
```suggestion
    <Layout title="Shop" hasFooter>
```

File: pages/specials.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import Layout from "../components/layout/Layout";
+
+const Specials = () => {
+  return (
+    <>
+      <Layout title={"Specials"} hasFooter={true}>
```
Comment (djfarly) [inline pages/specials.js:7]:
```suggestion
      <Layout title="Specials" hasFooter>
```

File: components/layout/Footer.jsx [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  .active {
+    background-color: var(--bg-color-main-white);
+    border: 3px solid var(--cta-color-main);
+  }
+
+  a {
+    width: var(--footer-link-width);
+    text-align: center;
+    line-height: calc(var(--footer-height) * 0.9);
+    height: calc(var(--footer-height) * 0.95);
```
Comment (djfarly) [inline components/layout/Footer.jsx:47]: This feels a little bit sketchy. I'd suggest not using line-height for layout purposes. For height i'd rather not set it or just set it to `100%` oder `var(--footer-height)`. Then solve vertical spacing with padding or margin.

File: components/layout/Layout.jsx [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+import React from "react";
+import Header from "./Header";
+import { Footer } from "./Footer";
+
+const Layout = ({ hasFooter, title, children }) => {
+  console.log(hasFooter);
+  return (
+    <>
+      <Header title={title} />
+      <StyledMain hasFooter={hasFooter}>{children} </StyledMain>
```
Comment (djfarly) [inline components/layout/Layout.jsx:11]: 1. `hasFooter` shorthand
2. there is an extra space after children. A single space in this case can actually end up in you DOM. Why it is there would be hard to track down.

```suggestion
      <StyledMain hasFooter>{children}</StyledMain>
```

**Review by djfarly — CHANGES_REQUESTED** [review]
Hej Volker,

some great progress! 👍

I've added some comments for some minor things that could be improved.

In general:
- Please add a title and description to your pull requests, this way reviewers have chance know what you were trying to do
- Please add a Vercel-Deploy to this GitHub-Repo, that way you and reviewers can see the running app in addition to the code

Anyways! Great work - keep on going. 🦦

File: components/layout/Footer.jsx (hunk 2 above)
Comment (VB2webD) [reply components/layout/Footer.jsx]: applied in final commit

File: components/layout/Footer.jsx (hunk 8 above)
Comment (VB2webD) [reply components/layout/Footer.jsx]: applied in final commit

**Pull request merged**